### PR TITLE
Deselect food, egg, and potion when hitting the escape key

### LINKED ISSUE #4772: Can't use Esc to cancel pet actions
When casting a spell, pressing the Esc key aborts the action. This, however, is not the case with pet actions like hatching and feeding. It'd be nice to be able to cancel one of those actions in progress by the same method.

### PR DESCRIPTION
**Pull request by TheHollidayInn**
This is a reboot solution to fix #4772 .

I added a global scope keydown event per this: http://stackoverflow.com/questions/24250263/capturing-document-level-keypress-events-in-angularjs . This solves an issue with the previous solution.

I also change the code to deselect all selected and added tests.

### CODE REVIEW

File: website/public/js/controllers/inventoryCtrl.js [hunk 1]
```diff
@@ -211,5 +211,25 @@ habitrpg.controller("InventoryCtrl",
           break;
       }
     };
+
+    $scope.$on("habit:keyup", function (e, keyEvent) {
+      if (keyEvent.keyCode == "27") {
+        $scope.deselectItem();
+      }
+    });
+ //$scope.listen(keyEvent)
```
Comment (crookedneighbor) [inline website/public/js/controllers/inventoryCtrl.js]: Intentional?

Comment (crookedneighbor) [inline website/public/js/controllers/inventoryCtrl.js]: Do we care if selectedWhatever has a value before setting it to null? Do we need these if statements?

File: test/spec/controllers/inventoryCtrlSpec.js [hunk 2]
```diff
@@ -89,4 +89,27 @@ describe('Inventory Controller', function() {
...
+      scope.deselectItem();
+      expect(scope.selectedPotion).to.eql(null);
+    });
+
+    it('deselects a egg', function(){
+      scope.chooseEgg('Cactus');
+      scope.deselectItem();
+      expect(scope.selectedEgg).to.eql(null);
+    });
+
```
Comment (crookedneighbor) [inline test/spec/controllers/inventoryCtrlSpec.js]: Out style for functions is to not leave a trailing line break like this or the one below.

**Comment by TheHollidayInn** [comment]
Updated!

**Comment by crookedneighbor** [comment]
Wonderful!

Tier awarded!

**Comment by TheHollidayInn** [comment]
Woot! :)

**Pull request merged**